calico_containers/calico_ctl/config.py

```diff
@@ -0,0 +1,192 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Usage:
+  calicoctl config felix <NAME> [<VALUE>|--remove] [--force]
+  calicoctl config bgp <NAME> [<VALUE>|--remove] [--force]
+  calicoctl config node bgp <NAME> [<VALUE>|--remove] [--force]
+
+Description:
+  Configure or show low-level component configuration for Felix and BGP.
+
+Options:
+ --remove  Remove the configuration entry.
+ --force   Force update of configuration entry even if key value is unknown,
+           or the value is not recognized as valid.
+
+Valid configuration:
+  Command         | <NAME>  | <VALUE>s
+------------------+---------+-----------------------------------------
+  config felix    | logging | none debug info warning error critical
+  config bgp      | logging | none debug info
+  config node bgp | logging | none debug info
+
+Warnings:
+  -  Changing the global BGP logging levels using the `calicoctl config bgp`
+     command may cause all BGP sessions to bounce potentially resulting in a
+     transient loss of service.  If you need to change the logging level for
+     BGP, it is recommended to change the levels on a per-node basis using
+     the `calicoctl config node bgp` command.
+"""
+import re
+import sys
+
+from etcd import EtcdKeyNotFound
+from pycalico.datastore import handle_errors
+from pycalico.datastore import CONFIG_PATH, BGP_HOST_PATH, BGP_GLOBAL_PATH
+
+from connectors import client
+from utils import print_paragraph
+from utils import hostname
+
+# Dictionaries providing look up between the configuration name, and a tuple
+# of (internal name, value regex string)
+FELIX_CONFIG_DATA = {
+    "logging": ("LogSeverityScreen", "none|debug|info|warning|error|critical")
+}
+BGP_CONFIG_DATA = {
+    "logging": ("LogSeverityScreen", "none|debug|info")
+}
+
+def validate_arguments(arguments):
+    """
+    Validate argument values:
+        <NAME>
+        <VALUE>
+
+    :param arguments: Docopt processed arguments
+    """
+    config_data = _get_config_data(arguments)
+    name = arguments["<NAME>"]
+    if name not in config_data:
+        print_paragraph("The configuration '%s' is not recognized as a "
+                        "valid option." % name)
+        if arguments["--force"]:
+            print_paragraph("The --force option is set.")
+            return
+        else:
+            print_paragraph("Use the --force option to override.")
+            sys.exit(1)
+
+    value = arguments.get("<VALUE>")
+    if value:
+        _, valid_values = config_data[name]
+        valid_values_re = re.compile(valid_values)
```

Comment:
With the current options a simple list of values would suffice.  I did regexes because I thought it would be a little more extensible... say for example you need a value to be an IP address.  Perhaps should encapsulate so you can define your own validator, but not worth it just at the moment.